src/main.rs

```diff
@@ -1226,6 +1227,20 @@ fn update_unix_map_with_peer(
     Ok(())
 }
 
+// Limit parallelism to no more than 8 threads
+// Anything higher slows down the program
+fn configure_parallelism() {
+    let num_cores = thread::available_parallelism()
+        .map(std::num::NonZeroUsize::get)
+        .unwrap_or(1);
+
+    let num_threads = std::cmp::min(MAX_THREADS, num_cores);
+    rayon::ThreadPoolBuilder::new()
+        .num_threads(num_threads)
+        .build_global()
+        .expect("Error configuring rayon thread pool");
```

Comment:
Instead of panicing, can we fall back to the current behavior, whatever that happens to be?